### DISCUSSION TITLE
New functionality proposal. Viewer comments in balloons.

### DISCUSSION THREAD

**Discussion by pauldim** [TVTower/TVTower#1160] [Ideas]
Viewer comments in balloons during a particular TV programme. 
This is my suggestion for expanding the game and the humorous behaviour of the viewers while watching. 
I propose to add a `<comment>comment</comment>` tag in the database which will cause a balloon to show in the viewers' screen with a viewer's commentary on a given programme, for, say, 10 minutes of game time. 
Example:
The movie A Clockwork Orange is broadcast and a viewer watching this movie comments on it with a balloon : "What a crazy movie".

**Comment by scr0llbaer9000** [comment]
Afair this is a feature Empire TV Tycoon has, right? Do you want to generally go into the direction of that game?

I think it would be a lot of work to add such comments to each movie.

Maybe there should be a general collection of viewer comments somewhere, and they would be applied to a movie more or less automatically depending on the genre and on defined target audiences. While still providing the option for an "override" for an individual movie here and there.

**Reply by pauldim** [reply]
Yes , this idea comes from this game. I think it is a great idea to expand the game, the atmosphere and the humour, which you like after all.
Yes, there will be a lot of work. Not every film needs to have audience commentary. If it's a matter of making the comments, I can take care of that later.

**Comment by GWRon** [comment]
The core functionality for comments/opinions for family members is already in the code base.

But of course it requires text instead of numbers then.

**Comment by scr0llbaer9000** [comment]
@pauldim maybe write down a bunch of more example comments you had in mind.

**Comment by pauldim** [comment]
It should also be noted that comments should be separate for each language.
Here are examples:


Star Wars - The Empire Strikes Back
"Not good! Darth Vader is Luke's father"

Star Wars - Return of Jedi
"What special effects! Great battle."

Feuer und Eis
I need to learn to ski.

Saving Private Ryan
"War is terrible!"

Inglourious Basterds
```
<Speech>
	<de>Dieser Film wurde in Deutschland gedreht.</de>
	<en>The director had an interesting idea for this film.</en>.
	<pl>Ten film jest podobny do parszywej dwunastki, tylko bardziej komediowy. He, he.</pl>
</Speech>
```

**Comment by pauldim** [comment]
The following examples, of course, the markup code would be different.
```
`<Name value="The Jungle Book">
<Speech value="Tiger force">

<Name value="Under The Shadow">
<Storyline value="A mother and her daughter's home is haunted by a mysterious evil in 1980s Tehran.">
<Speech value="Shideh">

<Name value="Goat">
<Storyline value="Reeling from a terrifying assault, a 19 year-old boy enrolls into college with his brother and pledges the same fraternity.">
<Speech value="Fraternity">

<Name value="War On Everyone">
<Storyline value="Two corrupt cops in New Mexico set out to blackmail and frame every criminal unfortunate enough to cross their path.">
<Year value="2016">
<Speech value="These guys are from Mexico">

<Name value="Soy Nero">
<Storyline value="Nero, a deported Mexican, returns illegally to the U.S. in search of his identity.">
<Speech value="A deported Mexican">

<Name value="Zero Days">
<Storyline value="Documentary detailing claims of American/Israeli jointly developed malware Stuxnet being deployed.">
<Speech value="What a virus">

<Name value="Hedi">
<Storyline value="Hedi a young man with great dream is struggling his way through social conventions in Tunisia.">
<Speech value="Great dream">

<Name value="Swiss Army Man">
<Storyline value="A hopeless man stranded in the wilderness befriends a dead body and together they go on a surreal journey to get home.">
<Speech value="Daniel Radcliffe!!">

<Name value="The 5th Wave">
<Storyline value="Four waves of increasingly deadly alien attacks have left most of Earth decimated.">
<Speech value="What a shit!">

<Name value="Allegiant">
<Storyline value="Tris must escape with Four beyond the wall that encircles Chicago.">
<Speech value="Tris go...">

<Name value="Deadpool">
<Storyline value="A former Special Forces operative turned mercenary is subjected to a rogue experiment.">
<Speech value="Shit. Did I leave the stove on?">

<Name value="How To Be Single">
<Storyline value="New York City is full of lonely hearts seeking the right match.">
<Speech value="I'm a single.">

<Name value="Zoolander 2">
<Storyline value="Derek and Hansel are lured into modeling again, in Rome, where they find themselves the target of a sinister conspiracy.">
<Speech value="What a heck?">

<Name value="Dirty Grandpa">
<Storyline value="Right before his wedding, an uptight guy is tricked into driving his grandfather to Florida for spring break.">
<Speech value="Robert De Niro is a nice grandpa.">

<Name value="Keanu">
<Storyline value="Friends hatch a plot to retrieve a stolen kitten by posing as drug dealers for a street gang.">
<Speech value="Sweet kitty..">

<Name value="Barbershop: The Next Cut">
<Storyline value="The crew at Calvin's Barbershop come together to bring some much needed change to their neighborhood.">
<Speech value="I have to go to the barber.">

<Name value="The Boss">
<Storyline value="A titan of industry is sent to prison after she's caught insider trading.">
<Speech value="Woman boss.">

<Name value="My Big Fat Greek Wedding 2">
<Storyline value="A Portokalos family secret brings the beloved characters back together for an even bigger and Greeker wedding.">
<Speech value="Kalimera">

<Name value="Gods of Egypt">
<Storyline value="Mortal hero Bek teams with the god Horus in an alliance against Set, the merciless god of darkness.">
<Speech value="Horus....Horus">

<Name value="Triple 9">
<Storyline value="A gang of criminals and corrupt cops plan the murder of a police officer in order to pull off their biggest heist yet across town.">
<Speech value="Criminals...">

<Name value="The Commune">
<Storyline value="A story about the clash between personal desires vs. solidarity and tolerance in a Danish commune in the 70's.">
<Speech value="Solidarity and tolerance.">

<Name value="Little Men">
<Storyline value="A new pair of best friends have their bond tested by their parents' battle over a dress shop lease.">
<Speech value="Don't argue.">

<Name value="Ma vie de Courgette">
<Storyline value="Courgette is an intriguing nickname for a 9 year old boy. Although his unique story is surprisingly universal.">
<Speech value="Cutie.">

<Name value="Kung Fu Panda 3">
<Storyline value="Continuing his legendary adventures of awesomeness, Po must face two hugely epic, but different threats.">
<Speech value="Haiiiii yaaaa">

<Name value="Bastille Day">
<Storyline value="A young con artist and former CIA agent embark on an anti-terrorist mission in France.">
<Speech value="I want to go Paris.">

<Name value="Ghostbusters">
<Storyline value="Following a ghost invasion of Manhattan, Erin, Abby, Jillian, and Patty band together to stop the otherworldly threat.">
<Year value="2016">
<Speech value="Who you gonna call? Ghostbusters. This is not real ghostbusters.">

<Name value="Blair Witch">
<Storyline value="After discovering a video, James and a group of friends head to the forest believed to be inhabited by the Blair Witch.">
<Speech value="What a nasty forest.">

<Name value="Suicide Squad">
<Storyline value="A secret government agency recruits some of the most dangerous incarcerated criminals to form a defensive task force.">
<Speech value="I love Margot Robbie.">`
```

**Reply by nittka** [reply]
I am a bit sceptic with respect to heavily movie-dependent comments.
* Without specifying which family member gives the comment, the result may be strange.
* Showing the same comment on every broadcast may turn this into a "non-feature"

On the other hand, generating generic feedback will not be easy to implement either (without being boring). You would have to provide *many* alteratives in the language property files and make sure the texts fit the context (how many broadcasts, genre fit, family-member-appropriate...). 

Do I understand your syntax correctly (separate file with comments, i.e. they are not part of the current programme database)?

**Comment by scr0llbaer9000** [comment]
Nice.

Though will also need different versions original / spoofed.

And we don't even have the full animations yet of the TV family as they were in Mad TV.

Also, if the missing room graphics could finally be included, the last "under construction" messages will be gone, and the game could be finally be regarded as "complete" for casual players. (Though open source software is never finished, of course.)

If you know anyone who can contribute those graphics, that would be much appreciated.

**Comment by pauldim** [comment]
We can still before the release of the new version add a hidden option to select original programs and ads and made-up (spoof)? You could then easily change the version of the game without changing the base. Hiding this option could be by password. Thanks.

**Reply by nittka** [reply]
You underestimate the changes required in the database code. I don't think this issue will be addressed for the current release.

**Reply by pauldim** [reply]
I don't know if you understand me? I would like it to take data from `<title original>` and `<description_original>`  instead of `<title>` and `<description>.` Just this. Or maybe ALSO that.

**Reply by nittka** [reply]
Yes, I understand. And the change needed in the database code (as well as the hidden, password-protected configuration possibility) is quite big. The scripts for transforming the database are out there.

**Reply by scr0llbaer9000** [reply]
It's just a few search-and-replace and file/folder operations. I can give you a script that does all that for now.

**Reply by pauldim** [reply]
Yes, I know how to do it. 
The problem is that with any change in the databases, editing, updating. you have to change it all over again. If it is so difficult well then there is no topic.

**Comment by pauldim** [comment]
I don't know if this is the moment to add images to TV shows anymore.  As luck would have it, all images for shows and series will be here in my github: [https://github.com/pauldim/MADTV/tree/main/IMG%20programmes](url). At the moment there are about 1800 of them. I don't know what resolution will be required, screen aspect ratio. They will all be here, as if you want to implement it, they can be useful.

**Reply by scr0llbaer9000** [reply]
They're copyrighted material so we can't include them (few exceptions are movies in the public domain, mostly very old ones; but even then it's not clear if a particular image from that movie downloaded from somewhere is in the public domain too).

Ideally we'd have spoofed versions of all of them. Maybe there's some AI tool for that.

With a variety of images we could then see how to best provide a way to configure and integrate them, the effect on memory and performance in the game, etc

Also would increase download size immensely, so maybe it would be better to make it a separate "media pack". Which would be a new project to support such "plugins".

So it's all not so easy.

**Reply by GWRon** [reply]
"plugins" is something we could tackle as one of the "bigger features" for next bigger release. But not especially for this thing here - but generally.
Would like to use physfs to "lay over" zip files over our filesystem - and thus allowing to "extend" or even override existing files this way (you "activate a zip" and it becomes a layer in a virtual file system). This way we should also be able to load media files etc.

Of course the problem of "thousands of assets" (what to load, keep in ram ...) has to be solved too. But the preparation of all these things could be used for various "asset packs" which can reside in various locations (preparation for linux-"user-folder"-support)
